fbot_manipulator_tools/fbot_manipulator_tools/saveArmPos.py

```diff
@@ -0,0 +1,175 @@
+import rclpy
+import yaml
+import os
+
+from rclpy.exceptions import ROSInterruptException
+from geometry_msgs.msg import PoseWithCovarianceStamped
+from state_machine.states import ServiceCallerState, PublisherState
+from interbotix_xs_msgs.srv import TorqueEnable
+import rclpy.wait_for_message
+from sensor_msgs.msg import JointState
+from collections import OrderedDict
+from rclpy.node import Node
+
+
+
+#TO DO: self.poses['poses']['ros__parameters']['targets'][pose_name] Line should have its values renamed to fit..
+# "poses" should be the current used node name and "targets" and "pose name" should follow the structure of the yaml file.
+'''
+Save the current pose in the specified topic to a yaml file.
+Created by Gabriel Dorneles on 2024-10-06.
+Port to ROS2 by Vitor Anello on 2025-05-7.
+Port to Use in Arm By Luiz Sparvoli on 2025-05-20.
+
+'''
+
+
+class OrderedDumper(yaml.SafeDumper):
+    '''
+    @brief Custom YAML dumper to handle OrderedDict.
+    '''
+    def represent_ordereddict(self, data):
+        return self.represent_dict(data.items())
+    
+class OrderedLoader(yaml.SafeLoader):
+    pass
+
+def construct_ordered_dict(loader, node):
+    '''
+    @brief Custom YAML loader to handle OrderedDict.b
+    '''
+    return OrderedDict(loader.construct_pairs(node))
+
+OrderedLoader.add_constructor(
+    yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG,
+    construct_ordered_dict
+)
+
+class PoseWriter (Node):
```

Comment:
This should have a different name than the original PoseWriter, I recommend:

```suggestion
class  ArmJointStateSaver(Node):
```